auth/middleware_test.go

```diff
@@ -0,0 +1,89 @@
+package auth
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/dmarro89/dare-db/logger"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMiddleware(t *testing.T) {
+	casbinAuth := NewCasbinAuth(modelPath, policyPath, Users{
+		"user1": {Roles: []string{"role1"}},
+		"user2": {Roles: []string{"role2"}},
+	})
+
+	userStore := NewUserStore()
+	authenticator := &JWTAutenticator{
+		usersStore: userStore,
+	}
+
+	middleware := &DareMiddleware{
+		authorizer:    casbinAuth,
+		authenticator: authenticator,
+		logger:        logger.NewDareLogger(),
+	}
+
+	handler := middleware.HandleFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+
+	// Test case where user1 is authorized for GET
+	req, err := http.NewRequest("GET", "/some-path", nil)
+	require.NoError(t, err)
+
+	token, err := middleware.authenticator.GenerateToken("user1")
```

Comment:
I suggest handling the error here with `require.NoError` as is required for such a critical case as token generation, so that the test execution stops on failure to generate a token.